### PR TITLE
Use PPX instead of camlp4

### PR DESCRIPTION
**Pull request by fxfactorial**

### CODE REVIEW

**Comment by ghost** [comment]
Thanks for your patches!

I just have a few comments:
- I see that you replaced the calls to `pkg-config` by hardcoded flags, I'm not sure that's very portable
- I'm not a big fan of committing lots of oasis generated lines in git. Having oasis as a build dependency when you clone the git repository seems acceptable to me. Any particular reason you committed the result of `oasis setup`?

**Comment by fxfactorial** [comment]
1) Ah, you're right, I can't build it as is on Ubuntu. Is there really no way to get this to work for oasis? 

``` shell
CCOpt:$(pkg-config --cflags libusb-1.0)
```

2) I don't mind them, since its just a preference and I had oasis dynamic give me headache before so i went with this static style approach.

**Comment by fxfactorial** [comment]
@diml Ah, I redid the history, now without the build changes.

File: src/USB.ml [hunk 1]
```diff
@@ -69,12 +68,12 @@ let transfer_error_message = function
...
-       | Transfer(err, fun_name) ->
-           Some(Printf.sprintf "USB transfer error: '%s' failed: %s" fun_name (transfer_error_message err))
-       | exn ->
-           None)
+      | Error(err, fun_name) ->
+        Some(Printf.sprintf "USB error: '%s' failed: %s" fun_name (error_message err))
+      | Transfer(err, fun_name) ->
+        Some(Printf.sprintf "USB transfer error: '%s' failed: %s" fun_name (transfer_error_message err))
+      | e ->
+        Some (Printexc.to_string e))
```
Comment (ghost) [inline src/USB.ml]: It should really be [None] here, [Printexc] takes care of trying the other registered printers. This line can actually create an infinite loop

Comment (fxfactorial) [inline src/USB.ml]: Fixed.

**Comment by ghost** [comment]
Unfortunately I don't think you can use pkg-config with oasis at the moment. There is an old ticket on the oasis bug tracker: https://forge.ocamlcore.org/tracker/?func=detail&aid=522&group_id=54&atid=294

I added a comment, appart from that it's ready to be merged.

**Comment by fxfactorial** [comment]
@diml Great, got that fix in.

**Comment by ghost** [comment]
Merged, thanks!

**Pull request merged**